server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -49,6 +51,8 @@ public class SegmentReplicationTargetService implements IndexEventListener {
 
     private final SegmentReplicationSourceFactory sourceFactory;
 
+    private final Map<ShardId, ReplicationCheckpoint> latestReceivedCheckpoint = new HashMap<>();
```

Comment:
It seems like key to this `Map` is always `replicaShard.shardId()`, meaning the Map will always have only one key-value pair. In this case, why not use a `Tuple` instead?

Using a `Map` here makes it look like each `SegmentReplicationTargetService` instance could receive checkpoints from multiple different shards, which is misleading.

Reply:
Ignore my comment above - `SegmentReplicationTargetService` is a singleton instance per _node_, and each _node_ can have multiple _shards_ on it. Therefore, a map is needed.